File: my_nifs.ex

defmodule :my_nifs do

  @on_load :load_nifs

  def load_nifs do
    :erlang.load_nif('./obj/development/Erlang_Nifs', 0)
  end

  def increment(_n) do
    raise "NIF increment/1 not implemented"
  end

  def negate(_n) do
    raise "NIF negate/1 not implemented"
  end

  def uppercase(_n) do
    raise "NIF uppercase/1 not implemented"
  end

  def uppercase_binary(_n) do
    raise "NIF uppercase_binary/1 not implemented"
  end

end

File: my_nifs.erl
-module(my_nifs).

-export([init/0, increment/1, negate/1, uppercase/1, uppercase_binary/1]).

-nifs([increment/1, negate/1, uppercase/1, uppercase_binary/1]).

-on_load(init/0).

init() ->
   erlang:load_nif("./obj/development/Erlang_Nifs", 0).

increment(_x) ->
   erlang:nif_error("NIF library not loaded").

negate(_x) ->
   erlang:nif_error("NIF library not loaded").

uppercase(_x) ->
   erlang:nif_error("NIF library not loaded").

uppercase_binary(_x) ->
   erlang:nif_error("NIF library not loaded").
